tests/unit/services/apigateway/test_handler_method_request.py

```diff
@@ -0,0 +1,288 @@
+import json
+
+import pytest
+
+from localstack.aws.api.apigateway import Method, Model, RequestValidator, RestApi
+from localstack.http import Request, Response
+from localstack.services.apigateway.models import MergedRestApi, RestApiDeployment
+from localstack.services.apigateway.next_gen.execute_api.api import RestApiGatewayHandlerChain
+from localstack.services.apigateway.next_gen.execute_api.context import (
+    InvocationRequest,
+    RestApiInvocationContext,
+)
+from localstack.services.apigateway.next_gen.execute_api.gateway_response import (
+    BadRequestBodyError,
+    BadRequestParametersError,
+)
+from localstack.services.apigateway.next_gen.execute_api.handlers import MethodRequestHandler
+from localstack.testing.config import TEST_AWS_ACCOUNT_ID, TEST_AWS_REGION_NAME
+
+
+@pytest.fixture
+def method_handler_chain() -> RestApiGatewayHandlerChain:
+    """Returns a dummy chain for testing."""
+    method_handler_chain = RestApiGatewayHandlerChain(request_handlers=[MethodRequestHandler()])
+    # Sets raise_on_error to allow capturing the Exceptions raised by the validator
+    method_handler_chain.raise_on_error = True
+    return method_handler_chain
```

Comment:
nit: should we use the handler chain with only handler, or should we directly instantiate the handler in the tests, so that we would not need to set `method_handler_chain.error = None` after every invocation?

In our Gateway, we instantiate a new handler chain for every incoming request.
We could maybe make the intent clearer by only testing the handler itself?

Reply:
I think you are right, I will see how I can make it cleaner by calling the handler itself! :smile:

Reply:
Very neat update 💯 I like that, I think we should follow the same pattern for new tests, with the invoker!